designer/custom-http-service-api/src/main/scala/pl/touk/nussknacker/ui/customhttpservice/services/ScenarioService.scala

```diff
@@ -0,0 +1,79 @@
+package pl.touk.nussknacker.ui.customhttpservice.services
+
+import pl.touk.nussknacker.engine.api.component.ProcessingMode
+import pl.touk.nussknacker.engine.api.deployment.{ProcessAction, ScenarioActionName}
+import pl.touk.nussknacker.engine.api.process._
+import pl.touk.nussknacker.engine.deployment.EngineSetupName
+import pl.touk.nussknacker.ui.customhttpservice.services.ScenarioService._
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import java.net.URI
+import java.time.Instant
+import scala.concurrent.Future
+
+trait ScenarioService {
+
+  def getLatestProcessesWithDetails(query: ScenarioQuery)(
+      implicit user: LoggedUser
+  ): Future[List[ScenarioWithDetails]]
+
+  def getLatestVersionForProcesses(query: ScenarioQuery, excludedUserNames: Set[String])(
```

Comment:
why does the `excludedUserNames` set cannot be a part of the query?

Reply:
It is not part of the scenario query (which filters data from scenarios table). It is about filtering process versions. If I added it to the ScenarioQuery, handling it in various places would be problematic. I created new similar wrapper for this value - ScenarioVersionQuery.

Reply:
looking from the perspective of this interface, the user (of the interface) wants to get sth. And can limit the response by using some kind of "query". Current underlying implementation fetches the data from two tables, but it doesn't mean we have to expose this information at the method level (by having two queries `query: ScenarioQuery, scenarioVersionQuery: ScenarioVersionQuery`)

Reply:
Ok, changed on SPI level. Separate queries for those 2 methods, but only one query per method.